File: TP2 python/python/Produk.py
class Produk:       # kelas produk
    # konstruktor untuk menginisialisasi atribut dasar produk
    def __init__(self, id="", nama="", harga=0):
        self.id = id            # menyimpan ID produk
        self.nama = nama        # menyimpan nama produk
        self.harga = harga      # menyimpan harga produk

    #getter untuk mengambil nilai
    def get_id(self): return self.id
    def get_nama(self): return self.nama
    def get_harga(self): return self.harga

File: TP2 CPP/CPP/Produk.cpp
#ifndef PRODUK_H
#define PRODUK_H
#include <iostream>
#include <string>
using namespace std;

//class produk sebagai parent
class Produk {
protected:
    string id;      // atribut ID Produk
    string nama;    // atribut nama Produk
    int harga;      // atribut harga Produk
public:
    // constuctor default
    Produk() {
        id = "";        // ID di isi string kosong
        nama = "";      // nama diisi string kosong
        harga = 0;      // isi di 0
    }

    // constuctor untuk mengisi atribut
    Produk(string id, string nama, int harga) {
        this->id = id;          // isi atribut ID
        this->nama = nama;      // isi atribut nama
        this->harga = harga;    // isi atribut harga
    }

    string getId() { return id; }           // getter untuk mengambil ID
    string getNama() { return nama; }       // getter untuk mengambil nama
    int getHarga() { return harga; }        // getter untuk mengambil harga
};

#endif
